src/binaryen-c.cpp

```diff
@@ -321,7 +321,10 @@ BinaryenExpressionRef BinaryenBlock(BinaryenModuleRef module, const char* name,
     auto id = noteExpression(ret);
     std::cout << "    expressions[" << id << "] = BinaryenBlock(the_module, ";
     traceNameOrNULL(name);
-    std::cout << ", children, " << numChildren << ", " << type << ");\n";
+    std::cout << ", children, " << numChildren << ", ";
+    if (type == BinaryenUndefined()) std::cout << "BinaryenUndefined()";
```

Comment:
For improved readability, it might be desirable to do this for all types in every traced function. That's some boring work, though, and I haven't yet managed to run the tests locally, so I decided not to touch this.